simde/arm/neon/addl.h

```diff
@@ -0,0 +1,335 @@
+  #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
+    return vaddl_high_s32(a, b);
+  #else
+    simde_int64x2_private r_;
+    simde_int32x4_private
+      a_ = simde_int32x4_to_private(a),
+      b_ = simde_int32x4_to_private(b);
+
+    size_t halfway = (sizeof(a_.values) / sizeof(a_.values[0])) / 2;
+    SIMDE_VECTORIZE
+    for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {
+      r_.values[i] = HEDLEY_STATIC_CAST(int64_t, a_.values[i + halfway]) + HEDLEY_STATIC_CAST(int64_t, b_.values[i + halfway]);
+    }
+
+    return simde_int64x2_from_private(r_);
+  #endif
+}
+#if defined(SIMDE_ARM_NEON_A64V8_ENABLE_NATIVE_ALIASES)
+  #undef vaddl_high_s32
+  #define vaddl_high_s32(a, b) simde_vaddl_high_s32((a), (b))
+#endif
+
+SIMDE_FUNCTION_ATTRIBUTES
+simde_uint16x8_t
+simde_vaddl_high_u8(simde_uint8x16_t a, simde_uint8x16_t b) {
+  #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
+    return vaddl_high_u8(a, b);
+  #else
+    simde_uint16x8_private r_;
+    simde_uint8x16_private
+      a_ = simde_uint8x16_to_private(a),
+      b_ = simde_uint8x16_to_private(b);
+
+    size_t halfway = (sizeof(a_.values) / sizeof(a_.values[0])) / 2;
+    SIMDE_VECTORIZE
+    for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {
+      r_.values[i] = HEDLEY_STATIC_CAST(uint16_t, a_.values[i + halfway]) + HEDLEY_STATIC_CAST(uint16_t, b_.values[i + halfway]);
+    }
+
+    return simde_uint16x8_from_private(r_);
+  #endif
+}
+#if defined(SIMDE_ARM_NEON_A64V8_ENABLE_NATIVE_ALIASES)
+  #undef vaddl_high_u8
+  #define vaddl_high_u8(a, b) simde_vaddl_high_u8((a), (b))
+#endif
+
+SIMDE_FUNCTION_ATTRIBUTES
+simde_uint32x4_t
+simde_vaddl_high_u16(simde_uint16x8_t a, simde_uint16x8_t b) {
+  #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
+    return vaddl_high_u16(a, b);
+  #else
+    simde_uint32x4_private r_;
+    simde_uint16x8_private
+      a_ = simde_uint16x8_to_private(a),
+      b_ = simde_uint16x8_to_private(b);
+
+    size_t halfway = (sizeof(a_.values) / sizeof(a_.values[0])) / 2;
+    SIMDE_VECTORIZE
+    for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {
+      r_.values[i] = HEDLEY_STATIC_CAST(uint32_t, a_.values[i + halfway]) + HEDLEY_STATIC_CAST(uint32_t, b_.values[i + halfway]);
+    }
+
+    return simde_uint32x4_from_private(r_);
+  #endif
+}
+#if defined(SIMDE_ARM_NEON_A64V8_ENABLE_NATIVE_ALIASES)
+  #undef vaddl_high_u16
+  #define vaddl_high_u16(a, b) simde_vaddl_high_u16((a), (b))
+#endif
+
+SIMDE_FUNCTION_ATTRIBUTES
+simde_uint64x2_t
+simde_vaddl_high_u32(simde_uint32x4_t a, simde_uint32x4_t b) {
+  #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
+    return vaddl_high_u32(a, b);
+  #else
+    simde_uint64x2_private r_;
+    simde_uint32x4_private
+      a_ = simde_uint32x4_to_private(a),
+      b_ = simde_uint32x4_to_private(b);
+
+    size_t halfway = (sizeof(a_.values) / sizeof(a_.values[0])) / 2;
+    SIMDE_VECTORIZE
+    for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {
```

Comment:
Is there a `arraylen` macro or something for this?

Reply:
Hm, I don't think there is, but adding it would be trivial, and would actually simplify a lot of the code. @nemequ want me to add it in?

Reply:
I just opened up #421 to discuss this.